homeassistant/components/stookwijzer/__init__.py

```diff
@@ -2,29 +2,68 @@
 
 from __future__ import annotations
 
+import logging
+
 from stookwijzer import Stookwijzer
 
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_LATITUDE, CONF_LOCATION, CONF_LONGITUDE, Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
 
 from .const import DOMAIN
+from .coordinator import StookwijzerCoordinator
 
 PLATFORMS = [Platform.SENSOR]
 
+_LOGGER = logging.getLogger(__name__)
+
+type StookwijzerConfigEntry = ConfigEntry[StookwijzerCoordinator]
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Stookwijzer from a config entry."""
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = Stookwijzer(
-        entry.data[CONF_LOCATION][CONF_LATITUDE],
-        entry.data[CONF_LOCATION][CONF_LONGITUDE],
+    client = Stookwijzer(
+        async_get_clientsession(hass),
+        entry.data[CONF_LATITUDE],
+        entry.data[CONF_LONGITUDE],
     )
+    coordinator = StookwijzerCoordinator(hass, client)
+    await coordinator.async_config_entry_first_refresh()
+
+    entry.runtime_data = coordinator
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
     return True
 
 
 async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
```suggestion
 async def async_unload_entry(hass: HomeAssistant, entry: StookwijzerConfigEntry) -> bool:
```

Reply:
Same as above, no specific use of entity needed here.